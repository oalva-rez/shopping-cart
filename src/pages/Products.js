import React from "react";
import { Link } from "react-router-dom";
import Header from "../components/Header";

export default function Catalog(props) {
  function displayProducts() {
    return props.products.map((product) => (
      <div className="product" key={product.id}>
        <Link to={`/products/${product.id}`}>
          <img src={product.image} alt={product.name} />
          <h3>{product.name}</h3>
          <p>{product.description}</p>
          <p>${product.price}</p>
        </Link>
      </div>
    ));
  }

  return (
    <div className="products">
      <div className="items">{displayProducts()}</div>
    </div>
  );
}
