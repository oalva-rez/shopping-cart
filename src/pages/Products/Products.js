import React from "react";
import { Link } from "react-router-dom";

export default function Catalog(props) {
  return (
    <div className="products">
      <h1 className="products-heading">All Products</h1>
      <div className="items">
        {props.products.map((product) => (
          <div className="product" key={product.id}>
            <Link to={`/products/${product.id}`} data-testid="product-link">
              <img src={product.image} alt={product.name} />
              <h3>{product.title}</h3>
              <p>${product.price}</p>
            </Link>
          </div>
        ))}
      </div>
    </div>
  );
}
