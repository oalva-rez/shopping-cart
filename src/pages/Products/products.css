.products-heading {
  text-align: center;
}
.items {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
  text-align: center;
  margin: 40px 50px;
  gap: 2rem;
}
.items img {
  height: 142px;
  width: 100px;
}

.product {
  padding: 20px;
}
.product a {
  text-decoration: none;
  color: #5b5b5b;
}

.product h3,
p {
  margin: 0;
}
.product h3 {
  font-weight: 400;
  font-size: 1rem;
  min-height: 40px;
}
.product p {
  margin-top: 7px;
}

.items {
  animation: fadeInAnimation ease 2s;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
}

@keyframes fadeInAnimation {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
@media (min-width: 1800px) {
  .products {
    width: 70vw;
    margin: auto;
  }
}
