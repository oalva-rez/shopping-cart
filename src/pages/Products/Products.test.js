import { render, screen } from "@testing-library/react";
import Products from "./Products";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import { BrowserRouter } from "react-router-dom";

const mockProductsAPICall = [
  {
    description: "Product description",
    id: 1,
    image: "https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg",
    price: 10,
    title: "Product 1",
  },
  {
    description: "Product 2 description",
    id: 2,
    image: "https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg",
    price: 11,
    title: "Product 2",
  },
];
const mockProducts = () => {
  return (
    <BrowserRouter>
      <Products products={mockProductsAPICall} />
    </BrowserRouter>
  );
};
describe("Products", () => {
  it("should match snapshot", () => {
    const { container } = render(mockProducts());
    expect(container).toMatchSnapshot();
  });
  it("should render product 1`s title", () => {
    render(mockProducts());
    expect(
      screen.getByRole("heading", { name: "Product 1" })
    ).toBeInTheDocument();
  });
  it("should render product 2`s title", () => {
    render(mockProducts());
    expect(
      screen.getByRole("heading", { name: "Product 2" })
    ).toBeInTheDocument();
  });
  it("has links to each product", () => {
    render(mockProducts());
    const links = screen.getAllByTestId("product-link");
    expect(links[0]).toHaveAttribute("href", "/products/1");
    expect(links[1]).toHaveAttribute("href", "/products/2");
  });
});
