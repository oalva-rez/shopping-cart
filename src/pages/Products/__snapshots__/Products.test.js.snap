// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products should match snapshot 1`] = `
<div>
  <div
    class="products"
  >
    <h1
      class="products-heading"
    >
      All Products
    </h1>
    <div
      class="items"
    >
      <div
        class="product"
      >
        <a
          data-testid="product-link"
          href="/products/1"
        >
          <img
            src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
          />
          <h3>
            Product 1
          </h3>
          <p>
            $
            10
          </p>
        </a>
      </div>
      <div
        class="product"
      >
        <a
          data-testid="product-link"
          href="/products/2"
        >
          <img
            src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
          />
          <h3>
            Product 2
          </h3>
          <p>
            $
            11
          </p>
        </a>
      </div>
    </div>
  </div>
</div>
`;
